### ISSUE TITLE
No video on iPad / mp4_to_annexb_write: zero size packet while processing frames

### ISSUE THREAD

**Issue by stueja** [kaltura/nginx-vod-module#907]
When trying to watch a video via hls, I get a blank, grey page without a video player on the iPad, and the message `mp4_to_annexb_write: zero size packet while processing frames, client: 192.168.1.2, server: vod.wtf.de, request: "GET /content/Tatort/Tatort_667_s2007e17.mp4/seg-1-v1.ts HTTP/1.1`.

On one hand, I am well aware that this might be an issue with the video itself, and I also read
 #856 and #769. 
On the other hand, the video plays very nicely in vlc, jwplayer, and the html5 video element in mobile Safari or Firefox.

I'd appreciate any hint on which route to go to debug this issue.

```
mediainfo /media/data/var/www/upload/otr/Tatort/Tatort_667_s2007e17.mp4 
General
Complete name                            : /media/data/var/www/upload/otr/Tatort/Tatort_667_s2007e17.mp4
Format                                   : MPEG-4
Format profile                           : Base Media / Version 2
Codec ID                                 : mp42 (isom/iso2/avc1)
File size                                : 263 MiB
Duration                                 : 1 h 40 min
Overall bit rate mode                    : Variable
Overall bit rate                         : 365 kb/s
Movie name                               : Tatort: Racheengel
Encoded date                             : UTC 2018-09-28 21:20:21
Tagged date                              : UTC 2018-09-28 21:20:21
Writing application                      : HandBrake 1.1.2 2018091000
Comment                                  : Fernsehfilm Deutschland 2007 Der Mord an einer Star-Autorin auf der Leipziger Buchmesse ist für die Hauptkommissare Ehrlicher und Kain kein fiktionaler Kriminalfall.

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Baseline@L1.3
Format settings                          : 3 Ref Frames
Format settings, CABAC                   : No
Format settings, ReFrames                : 3 frames
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 1 h 40 min
Bit rate                                 : 198 kb/s
Width                                    : 320 pixels
Height                                   : 176 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 25.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.140
Stream size                              : 143 MiB (54%)
Writing library                          : x264 core 155 r2917 0a84d98
Encoding settings                        : cabac=0 / ref=3 / deblock=1:0:0 / analyse=0x1:0x111 / me=hex / subme=7 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / me_range=16 / chroma_me=1 / trellis=1 / 8x8dct=0 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=5 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=0 / weightp=0 / keyint=250 / keyint_min=25 / scenecut=40 / intra_refresh=0 / rc_lookahead=40 / rc=crf / mbtree=1 / crf=22.0 / qcomp=0.60 / qpmin=0 / qpmax=69 / qpstep=4 / vbv_maxrate=768 / vbv_bufsize=2000 / crf_max=0.0 / nal_hrd=none / filler=0 / ip_ratio=1.40 / aq=1:1.00
Encoded date                             : UTC 2018-09-28 21:20:21
Tagged date                              : UTC 2018-09-28 21:20:21
Color range                              : Limited
Color primaries                          : BT.601 PAL
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.601
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 1 h 40 min
Bit rate mode                            : Variable
Bit rate                                 : 160 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 116 MiB (44%)
Title                                    : Stereo
Default                                  : Yes
Alternate group                          : 1
Encoded date                             : UTC 2018-09-28 21:20:21
Tagged date                              : UTC 2018-09-28 21:20:21



```

The video was transcoded via `HandBrakeCLI --arate 44.1 --optimize --keep-display-aspect --preset "Apple 240p30" -i /mnt/media/data/MyVideos/TVShows/Tatort/Tatort_667_s2007e17.avi -o /mnt/media/data/var/www/upload/otr/Tatort/Tatort_667_s2007e17.mp4`

Mediainfo of the originating avi:
```
mediainfo /media/data/MyVideos/TVShows/Tatort/Tatort_667_s2007e17.avi 
General
Complete name                            : /media/data/MyVideos/TVShows/Tatort/Tatort_667_s2007e17.avi
Format                                   : AVI
Format/Info                              : Audio Video Interleave
File size                                : 675 MiB
Duration                                 : 1 h 40 min
Overall bit rate                         : 936 kb/s
Movie name                               : Tatort: Racheengel
Encoded by                               : www.onlinetvrecorder.com
Writing application                      : Lavf54.17.100
Comment                                  : Fernsehfilm Deutschland 2007 Der Mord an einer Star-Autorin auf der Leipziger Buchmesse ist für die Hauptkommissare Ehrlicher und Kain kein fiktionaler Kriminalfall.

Video
ID                                       : 0
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L3
Format settings                          : CABAC / 4 Ref Frames
Format settings, CABAC                   : Yes
Format settings, ReFrames                : 4 frames
Codec ID                                 : H264
Duration                                 : 1 h 40 min
Bit rate                                 : 731 kb/s
Width                                    : 720 pixels
Height                                   : 576 pixels
Display aspect ratio                     : 16:9
Frame rate                               : 25.000 FPS
Standard                                 : PAL
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.070
Stream size                              : 527 MiB (78%)
Writing library                          : x264 core 125 r2200 999b753
Encoding settings                        : cabac=1 / ref=3 / deblock=1:-1:-1 / analyse=0x3:0x113 / me=hex / subme=7 / psy=1 / fade_compensate=0.10 / psy_rd=1.00:0.15 / mixed_ref=1 / me_range=16 / chroma_me=1 / trellis=1 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-3 / threads=6 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=3 / b_pyramid=2 / b_adapt=2 / b_bias=0 / direct=3 / weightb=1 / open_gop=0 / weightp=0 / keyint=250 / keyint_min=25 / scenecut=40 / intra_refresh=0 / rc_lookahead=60 / rc=crf / mbtree=1 / crf=23.0000 / qcomp=0.60 / qpmin=0 / qpmax=69 / qpstep=4 / ip_ratio=1.40 / aq=2:1.00
Color range                              : Limited
Color primaries                          : BT.601 PAL
Transfer characteristics                 : BT.470 System B/G
Matrix coefficients                      : BT.470 System B/G

Audio
ID                                       : 1
Format                                   : MPEG Audio
Format version                           : Version 1
Format profile                           : Layer 3
Format settings                          : Joint stereo / MS Stereo
Codec ID                                 : 55
Codec ID/Hint                            : MP3
Duration                                 : 1 h 40 min
Bit rate mode                            : Constant
Bit rate                                 : 192 kb/s
Channel(s)                               : 2 channels
Sampling rate                            : 48.0 kHz
Compression mode                         : Lossy
Stream size                              : 138 MiB (21%)
Alignment                                : Aligned on interleaves
Interleave, duration                     : 24  ms (0.60 video frame)

```

**Comment by erankor** [comment]
Please provide a URL to the MP4 file

**Comment by stueja** [comment]
Please have a look here:
https://drive.google.com/open?id=16a4zBvPDsSfbkFg3CRWJzvgnqJxmk5r3

**Comment by stueja** [comment]
Does that suffice? Or would you prefer a different download location?

**Comment by erankor** [comment]
Sorry, checked it out only now, but it is playing fine for me... maybe you sent the wrong file?
I got Tatort_662_s2007e12.mp4 while you pasted above the mediainfo of Tatort_667_s2007e17.mp4.
Also, please avoid sending files over Drive, as it's blocked in our org.

**Comment by stueja** [comment]
Oh, wow. Thanks for hinting me that I uploaded the wrong file to the wrong location ;) 
I put the correct file here:
https://mega.nz/#!nDBgHQgA

**Comment by erankor** [comment]
It asks me for a decryption key...

**Comment by stueja** [comment]
Great Odin's raven. Sorry for that. Try that one here.
https://mega.nz/#!nDBgHQgA!kmpuMq4R4XfkTUyhemQxFaPiz5XFZi2D2OlmkGR04gU

**Comment by erankor** [comment]
No go... getting an error...

**Comment by stueja** [comment]
So, I did additional tests, trying to reproduce the process, with that very file and with other files, I used current and older versions of HandBrake to encode, and, what can I say? I can say, sorry, the problem has glitched away. After all that hassle I can't reproduce the problem.
The only thing I can think of is that maybe iOS was updated in between our tests.
Again, sorry for that noise and thanks for your patience.
